File: competition/01.周赛/334.场周赛/03.求出最多标记下标.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxNumOfMarkedIndices = function (nums) {
  nums.sort((a, b) => a - b)
  const n = nums.length
  let count = 0
  const map = {}
  for (let i = 0; i < n; i++) {
    for (let j = i + 1; j < n; j++) {
      //   console.log(map, 'map')
      if (!map[i] && !map[j] && nums[i] * 2 <= nums[j]) {
        map[i] = true
        map[j] = true
        count += 2
        console.log(map, '--')
      }
    }
  }
  console.log(count)
  return count
}

// maxNumOfMarkedIndices([3, 5, 2, 4])
maxNumOfMarkedIndices([9, 2, 5, 4])
// maxNumOfMarkedIndices([7, 6, 8])

File: competition/01.周赛/334.场周赛/03.求出最多标记下标.java
class Solution {
    public int maxNumOfMarkedIndices(int[] nums) {
        Arrays.sort(nums);
        int n = nums.length;
        int left = 0, right = n / 2 + 1;
        while (left + 1 < right) {
            int mod = (left + right) >> 1;
            if (check(nums,mod)) left = mod;
            else right = mod;
        }
        return left * 2;
    }
      public boolean check(int[] nums,int k) {
        for (int i = 0; i < k; i++) {
            if (nums[i] * 2 > nums[nums.length - k + i]) return false;
        }
        return  true;
    }
}